src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/OnesComplementInstruction.cs

```diff
@@ -1,175 +0,0 @@
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object obj = frame.Pop();
-                if (obj == null)
-                {
-                    frame.Push(null);
-                }
-                else
-                {
-                    frame.Push(~(uint)obj);
-                }
-                return 1;
-            }
-        }
-
-        private sealed class OnesComplementUInt64 : OnesComplementInstruction
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object obj = frame.Pop();
-                if (obj == null)
-                {
-                    frame.Push(null);
-                }
-                else
-                {
-                    frame.Push(~(ulong)obj);
-                }
-                return 1;
-            }
-        }
-
-        private sealed class OnesComplementByte : OnesComplementInstruction
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object obj = frame.Pop();
-                if (obj == null)
-                {
-                    frame.Push(null);
-                }
-                else
-                {
-                    frame.Push(unchecked((byte)(~(byte)obj)));
-                }
-                return 1;
-            }
-        }
-
-        private sealed class OnesComplementSByte : OnesComplementInstruction
-        {
-            public override int Run(InterpretedFrame frame)
-            {
-                object obj = frame.Pop();
-                if (obj == null)
-                {
-                    frame.Push(null);
-                }
-                else
-                {
-                    frame.Push((sbyte)(~(sbyte)obj));
-                }
-                return 1;
-            }
-        }
-
-        public static Instruction Create(Type type)
-        {
-            Debug.Assert(!type.IsEnum);
-            switch (type.GetNonNullableType().GetTypeCode())
-            {
-                case TypeCode.SByte: return s_SByte ?? (s_SByte = new OnesComplementSByte());
-                case TypeCode.Int16: return s_Int16 ?? (s_Int16 = new OnesComplementInt16());
-                case TypeCode.Int32: return s_Int32 ?? (s_Int32 = new OnesComplementInt32());
-                case TypeCode.Int64: return s_Int64 ?? (s_Int64 = new OnesComplementInt64());
-                case TypeCode.Byte: return s_Byte ?? (s_Byte = new OnesComplementByte());
-                case TypeCode.UInt16: return s_UInt16 ?? (s_UInt16 = new OnesComplementUInt16());
-                case TypeCode.UInt32: return s_UInt32 ?? (s_UInt32 = new OnesComplementUInt32());
-                case TypeCode.UInt64: return s_UInt64 ?? (s_UInt64 = new OnesComplementUInt64());
-                default:
-                    throw Error.ExpressionNotSupportedForType("OnesComplement", type);
```

Comment:
That exception was unreachable though, and the boolean case is still only reachable through Not expressions.
I'll check there's a test that attempts one's complement on booleans, and add it if it's absent, when I'm not on phone